storage/interceptors/broker_notifications_interceptor.go

```diff
@@ -52,15 +53,58 @@ func NewBrokerNotificationsInterceptor() *NotificationsInterceptor {
 			details := make(objectDetails, objects.Len())
 			for i := 0; i < objects.Len(); i++ {
 				broker := objects.ItemAt(i).(*types.ServiceBroker)
+				services := broker.Services
+				if len(services) == 0 {
+					var err error
+					services, err = loadServicesFromStorage(ctx, broker, repository)
+					if err != nil {
+						return nil, err
+					}
+				}
 				details[broker.ID] = &BrokerAdditional{
-					Services: broker.Services,
+					Services: services,
 				}
 			}
 			return details, nil
 		},
 	}
 }
 
+func loadServicesFromStorage(ctx context.Context, broker *types.ServiceBroker, repository storage.Repository) ([]*types.ServiceOffering, error) {
```

Comment:
You can use https://github.com/Peripli/service-manager/blob/9bcb485ef331994537ec3a7e82a564a7d119a47f/storage/catalog/loader.go